src/middleware/passport-auth.ts

```diff
@@ -0,0 +1,168 @@
+import { createPrivateKey } from 'node:crypto';
+
+import passport from 'passport';
+import { Issuer, Strategy as OpenIdStrategy, TokenSet, UserinfoResponse } from 'openid-client';
+import { Strategy as GoogleStrategy } from 'passport-google-oauth20';
+import { Strategy as JWTStrategy, ExtractJwt } from 'passport-jwt';
+import { Repository } from 'typeorm';
+
+import { logger } from '../utils/logger';
+import { User } from '../entities/user';
+
+const readPrivateKey = (privateKey: string) => {
+    return createPrivateKey({ key: Buffer.from(privateKey, 'base64'), type: 'pkcs8', format: 'der' });
+};
+
+export const initPassport = async (userRepository: Repository<User>): Promise<void> => {
+    passport.use(
+        'jwt',
+        new JWTStrategy(
+            {
+                jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
+                secretOrKey: process.env.JWT_SECRET!
+            },
+            async (jwtPayload, done) => {
+                logger.info('request authenticated with JWT');
+                const user = await userRepository.findOneBy({ id: jwtPayload?.user?.id });
```

Comment:
I decided to always reload the user from the db, as I can't think of a good reason not to.

Reply:
Could bury the user in a session and get it from there but honestly it wouldn't save any overhead or time.  So this is fine.  If a user is disabled it means we can lock out a user more quickly rather than waiting for their session to expire.

Reply:
So once you're authed, `req.user` is set with the user object from here and available in any authed route.